src/code/z_horse.c

```diff
@@ -127,7 +127,7 @@ void func_8006D0EC(GlobalContext* globalCtx, Player* player) {
             }
         }
     } else if (!Flags_GetEventChkInf(0x18)) {
-        if ((DREG(1) == 0) && (globalCtx->sceneNum == SCENE_SOUKO) && (gSaveContext.nightFlag != 0)) {
+        if ((DREG(1) == 0) && (globalCtx->sceneNum == SCENE_SOUKO) && (gSaveContext.nightFlag)) {
```

Comment:
`nightFlag` isn't really a boolean (like `linkAge`), so I think we should leave these checks as `== 0`/`!= 0`/`== 1` for now. Eventually, we will probably have `IS_DAY`/`IS_NIGHT` macros to handle it (including the `== 1` checks), so we don't need to worry about treating it like a boolean.

Reply:
in kankyo where its set at least, it is def just true or false for currently being night. though they do check explicitly for "true" alot 

will it be possible to have a macro that uses both methods of checking?

Reply:
`IS_DAY` and `IS_NIGHT` can handle everything, just like `LINK_IS_CHILD` and `LINK_IS_ADULT` will once they are adjusted.

Right now, my idea is to use `LINK_IS_CHILD` for `linkAge == 0` and `LINK_IS_ADULT` for `linkAge == 1`. Then we can use `!LINK_IS_CHILD` for `linkAge != 0`, and `!LINK_IS_ADULT` for `linkAge != 1`. With the same pattern for day/night stuff ofc

Reply:
ah yea that makes sense, alright.
though thats kind of unfortunate for the age stuff. `!LINK_IS_CHILD` is way more common than `LINK_IS_ADULT` and the macro saying adult is a bit easier on the eyes. but thats just how it is i guess

Reply:
I just realized I inverted child and adult in this explanation, but you get the idea. I agree it's unfortunate, but chances are they were just inconsistent about it and I don't think there is much we can do about that.

Reply:
To some extent, it makes sense to check for "not being adult" more often, if we consider that adult is the "normal" age (or the main one)